### PR TITLE
Clean up ConstantPoolObjectSlotIterator class

### PR DESCRIPTION
**Pull request by LinHu2016**
Use consistent data types and code formatting

### CODE REVIEW

File: runtime/gc_structs/ConstantPoolObjectSlotIterator.hpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ class GC_ConstantPoolObjectSlotIterator
 	{
 		_cpEntryTotal = _cpEntryCount;
 		if(_cpEntryCount) {
-			_cpDescriptionSlots = SRP_PTR_GET(&clazz->romClass->cpShapeDescription, U_32 *);
+			_cpDescriptionSlots = SRP_PTR_GET(&clazz->romClass->cpShapeDescription, uint32_t *);
 			_cpDescriptionIndex = 0;
 		}
 	};
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolObjectSlotIterator.hpp:70]: no need for semicolon

File: runtime/gc_structs/ConstantPoolObjectSlotIterator.hpp [hunk 2]
```diff
@@ -64,7 +64,7 @@ class GC_ConstantPoolObjectSlotIterator
 	{
 		_cpEntryTotal = _cpEntryCount;
 		if(_cpEntryCount) {
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolObjectSlotIterator.hpp:66]: add space in 'if('

File: runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp [hunk 3]
```diff
@@ -78,7 +78,7 @@ GC_ConstantPoolObjectSlotIterator::nextSlot() {
 		}
 
 		/* Adjust the CP slot and description information */
-		_cpEntry = (j9object_t *) (((U_8 *) _cpEntry)
+		_cpEntry = (j9object_t *) (((uint8_t *) _cpEntry)
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp:81]: I tend to not leave spece between cast operatior and argument, but not sure what coding standard say (and what's more common) @dmitripivkine 

(j9object_t *)(((uint8_t *)_cpEntry)

Comment (dmitripivkine) [reply runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp:81]: yes, agreed

Comment (dmitripivkine) [reply runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp:81]: also we can remove unnecessary space lines 61, 64, 67

**Comment by amicic** [comment]
jenkins compile win jdk8

**Comment by amicic** [comment]
jenkins compile win jdk8

**Comment by amicic** [comment]
jenkins compile aix jdk8

**Review by amicic — APPROVED** [review]

**Pull request merged**